crates/polars-ops/src/frame/pivot/mod.rs

```diff
@@ -192,112 +192,133 @@ fn pivot_impl(
-                            Count => groups.group_count().into_series(),
-                            Expr(ref expr) => {
-                                let name = expr.root_name()?;
-                                let mut value_col = value_col.clone();
-                                value_col.rename(name);
-                                let tmp_df = DataFrame::new_no_checks(vec![value_col]);
-                                let mut aggregated = expr.evaluate(&tmp_df, &groups)?;
-                                aggregated.rename(value_col_name);
-                                aggregated
-                            }
-                        },
+            use PivotAgg::*;
+            let value_agg = unsafe {
+                match &agg_fn {
+                    None => match value_col.len() > groups.len() {
+                        true => polars_bail!(ComputeError: "found multiple elements in the same group, please specify an aggregation function"),
+                        false => value_col.agg_first(&groups),
                     }
-                };
-
-                let headers = column_agg.unique_stable()?.cast(&DataType::String)?;
-                let mut headers = headers.str().unwrap().clone();
-                if values.len() > 1 {
-                    headers = headers.apply_values(|v| Cow::from(format!("{value_col_name}{sep}{column_column_name}{sep}{v}")))
+                    Some(agg_fn) => match agg_fn {
+                        Sum => value_col.agg_sum(&groups),
+                        Min => value_col.agg_min(&groups),
+                        Max => value_col.agg_max(&groups),
+                        Last => value_col.agg_last(&groups),
+                        First => value_col.agg_first(&groups),
+                        Mean => value_col.agg_mean(&groups),
+                        Median => value_col.agg_median(&groups),
+                        Count => groups.group_count().into_series(),
+                        Expr(ref expr) => {
+                            let name = expr.root_name()?;
+                            let mut value_col = value_col.clone();
+                            value_col.rename(name);
+                            let tmp_df = DataFrame::new_no_checks(vec![value_col]);
+                            let mut aggregated = expr.evaluate(&tmp_df, &groups)?;
+                            aggregated.rename(value_col_name);
+                            aggregated
+                        }
+                    },
                 }
+            };
 
-                let n_cols = headers.len();
-                let value_agg_phys = value_agg.to_physical_repr();
-                let logical_type = value_agg.dtype();
+            let headers = column_agg.unique_stable()?.cast(&DataType::String)?;
+            let mut headers = headers.str().unwrap().clone();
+            if values.len() > 1 {
+                headers = headers.apply_values(|v| Cow::from(format!("{value_col_name}{sep}{v}")))
```

Comment:
instead of treating each column independently, they're now all grouped over together. so, the extra `{sep}{column_column_name}` doesn't add anything anymore other than verbosity, and can be removed